### ISSUE TITLE
The site cross-read guard covers two of the three ways CI discovers a test

### ISSUE THREAD

**Issue by zernie** [zernie/agent-lint#225]
Follow-up to #223, which closed the hole and hardened the guard through six rounds of review. Two gaps are left open **on purpose** — both widen the guard into new territory rather than fix a false claim, and the PR had reached the point where each round found one more dimension.

## 1. The `harness` job's discovery is a third mechanism

The guard derives what to scan from the vitest projects and jest's `testMatch`. The skipped `harness` job runs neither:

```yaml
# .github/workflows/ci.yml — the harness job
node dist/cli.js test --min=14
```

`handleRunScripts` discovers `**/*.harness.{mjs,cjs,js,mts,cts,ts}` across `examples/`, `.claude/` and elsewhere — locations no config glob names. A harness script reading a site fixture is invisible to the guard **and** skipped by a site-only diff, which is the original failure mode intact for that suite.

Fix shape: the scan should take the harness command's own discovery glob, the same way it now takes the configs' — the value, not a copy of it. Worth checking whether `discoverScripts` can be called directly rather than having its glob restated, since a restated glob is the thing this guard keeps getting wrong.

## 2. An aliased filesystem reader is not recognised

The AST walk classifies a call by the callee's name, so a renaming import defeats it:

```ts
import { readFileSync as read } from "node:fs";
read("../site/x.json", "utf8");   // not reported
```

**Measured: zero occurrences in the repo today** (`readFileSync as` / `readdirSync as` / `existsSync as` across `src/`, `scripts/`, `eslint-rules/`, `test/`), so this is a gap, not a live defect — which is why it is here and not in the PR.

Fix shape: resolve imported bindings to their original export before matching, instead of recording a list of aliases. A list is another spelling to maintain, and spellings are what cost that PR six rounds.

## Why these two are here rather than in #223

Round after round in one file is the signal to stop patching and look at the shape — the PR's last change already replaced text matching with a real parse for that reason. These two are not more spellings of the same check; they are a third discovery source and a binding-resolution question. Both deserve their own diff.

A third finding from the same round **was** fixed in the PR, because it falsified a claim the PR made rather than widening it: the consumer test rendered the chip component directly, so deleting `<HeroChip />` from `Hero` left it green.
